sky/check.py

```diff
@@ -20,44 +21,91 @@
 PARTY_POPPER_EMOJI = '\U0001F389'  # Party popper unicode
 
 
-def check(
+# Declaring CloudCapability as a subclass of str
+# allows it to be JSON serializable.
+class CloudCapability(str, enum.Enum):
+    # Compute capability.
+    COMPUTE = 'compute'
+    # Storage capability.
+    STORAGE = 'storage'
+
+
+ALL_CAPABILITIES = [CloudCapability.COMPUTE, CloudCapability.STORAGE]
+
+
+def check_capabilities(
     quiet: bool = False,
     verbose: bool = False,
     clouds: Optional[Iterable[str]] = None,
-) -> List[str]:
+    capabilities: Optional[List[CloudCapability]] = None,
+) -> Dict[str, List[CloudCapability]]:
     echo = (lambda *_args, **_kwargs: None
            ) if quiet else lambda *args, **kwargs: click.echo(
                *args, **kwargs, color=True)
     echo('Checking credentials to enable clouds for SkyPilot.')
-    enabled_clouds = []
-    disabled_clouds = []
+    if capabilities is None:
+        capabilities = ALL_CAPABILITIES
+    assert capabilities is not None
+    enabled_clouds: Dict[str, List[CloudCapability]] = {}
+    disabled_clouds: Dict[str, List[CloudCapability]] = {}
+
+    def check_credentials(
```

Comment:
discussed offline - we will look at cleaning up these functions and pushing down the cloud capability casing into global_user_state